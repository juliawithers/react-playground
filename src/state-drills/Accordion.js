import React, { Component } from 'react'
import ReactDOM from 'react-dom';

export default class Accordion extends Component {
    static defaultProps = {
        sections: []
    };
    state = {
        currentId: null,
    }

    setId = (secId) => {
            this.setState({currentId: secId})
    }

    renderText(section,id,currentId){
        return(
            <li className="list-info" key={id}>
                <button
                type='button' 
                onClick={() => this.setId(id)}
                >
                    {section.title} 
                </button>
            
                {(currentId === id && 
                <p>
                    {section.content}
                </p>)}           
            </li>  
        )
    }

    render(){
        const { sections } = this.props
        const { currentId } = this.state
        return(
            <ul>
                {sections.map((section,id)=>
                this.renderText(section,id,currentId)
                )}
            
            </ul>
        )
    } 
}