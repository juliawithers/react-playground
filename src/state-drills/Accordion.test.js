import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Accordion from './Accordion'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`Accordion Component`,() =>{
    // snapshot tests

    const sectionsProp= [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]
      
      it(`renders empty given no sections`,() =>{
          const wrapper = shallow(<Accordion />)
          expect(toJson(wrapper)).toMatchSnapshot()
      })
      it(`renders the buttons only by default`,()=>{
          const wrapper = shallow(<Accordion sections={sectionsProp} />)
          expect(toJson(wrapper)).toMatchSnapshot()
      })
      it(`opens the content when button is clicked`,()=>{
          const wrapper = shallow(<Accordion sections ={sectionsProp}/>)
        //   console.log('>>> WRAPPER <<<')
        //   console.log(wrapper.debug())
        //   console.log('>>> FIND(BUTTON) <<<')
        //   console.log(wrapper.find('button').debug())
        //   console.log(wrapper.find('button').at(1).debug())
          wrapper.find('button').at(1).simulate('click')
          expect(toJson(wrapper)).toMatchSnapshot()
      })
      it(`opens one section at a time`,()=>{
        const wrapper = shallow(<Accordion sections ={sectionsProp}/>)
          wrapper.find('button').at(1).simulate('click')
          wrapper.find('button').at(2).simulate('click')
          expect(toJson(wrapper)).toMatchSnapshot()
      })
})
