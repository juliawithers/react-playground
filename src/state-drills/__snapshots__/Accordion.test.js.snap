// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component opens one section at a time 1`] = `
<ul>
  <li
    className="list-info"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="list-info"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="list-info"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion Component opens the content when button is clicked 1`] = `
<ul>
  <li
    className="list-info"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="list-info"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="list-info"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no sections 1`] = `<ul />`;

exports[`Accordion Component renders the buttons only by default 1`] = `
<ul>
  <li
    className="list-info"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="list-info"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="list-info"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;
